cmd/fluxctl/lock_cmd.go

```diff
@@ -6,37 +6,52 @@ import (
 	"github.com/weaveworks/flux/update"
 )
 
-type serviceLockOpts struct {
+type controllerLockOpts struct {
 	*rootOpts
-	service string
+	namespace  string
+	controller string
 	outputOpts
 	cause update.Cause
+
+	// Deprecated
+	service string
 }
 
-func newServiceLock(parent *rootOpts) *serviceLockOpts {
-	return &serviceLockOpts{rootOpts: parent}
+func newControllerLock(parent *rootOpts) *controllerLockOpts {
+	return &controllerLockOpts{rootOpts: parent}
 }
 
-func (opts *serviceLockOpts) Command() *cobra.Command {
+func (opts *controllerLockOpts) Command() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "lock",
-		Short: "Lock a service, so it cannot be deployed.",
+		Short: "Lock a controller, so it cannot be deployed.",
 		Example: makeExample(
-			"fluxctl lock --service=helloworld",
+			"fluxctl lock --controller=deployment/helloworld",
 		),
 		RunE: opts.RunE,
 	}
 	AddOutputFlags(cmd, &opts.outputOpts)
 	AddCauseFlags(cmd, &opts.cause)
+	cmd.Flags().StringVarP(&opts.namespace, "namespace", "n", "default", "Controller namespace")
+	cmd.Flags().StringVarP(&opts.controller, "controller", "c", "", "Controller to lock")
+
+	// Deprecated
 	cmd.Flags().StringVarP(&opts.service, "service", "s", "", "Service to lock")
+	cmd.Flags().MarkHidden("service")
```

Comment:
Isn't there a `MarkDeprecated`?

Reply:
There is, but it just prints a warning message and then continues as normal. This meant I had to manually test to see if the arg was supplied and then error out, so I just used `MarkHidden` + the normal error return path to deliver the message.

Reply:
Got it :+1: